android/app/src/main/java/org/cryptimeleon/incentive/app/data/CryptoRepository.kt

```diff
@@ -146,7 +145,15 @@ class CryptoRepository(
         val pp = IncentivePublicParameters(jsonConverter.deserialize(remotePP))
         val incentiveSystem = IncentiveSystem(pp)
         val providerPublicKey = ProviderPublicKey(jsonConverter.deserialize(remotePPK), pp)
-        val userKeyPair = incentiveSystem.generateUserKeys()
+        val userPreKeyPair = incentiveSystem.generateUserPreKeyPair()
+        val signatureResponse =
+            cryptoApiService.retrieveGenesisSignatureFor(jsonConverter.serialize(userPreKeyPair.pk.upk.representation))
```

Comment:
publicly accessible variables?

Reply:
What do you mean by that?

Reply:
I was confused why pk and upk in UserPreKeyPair look like they are publicly accessible

Reply:
Maybe you are confused by kotlins syntactic sugar? In kotlin you can use *.pk instead of *.getPk()